core/src/main/scala/org/http4s/otel4s/middleware/ClientMiddleware.scala

```diff
@@ -151,36 +148,19 @@ object ClientMiddleware {
             trace = res.trace
             traceHeaders <- Resource.eval(Tracer[F].propagate(Headers.empty)).mapK(trace)
             newReq = req.withHeaders(traceHeaders ++ req.headers)
-            resp <- poll(client.run(newReq)).guaranteeCase { outcome =>
-              (outcome match {
-                case Outcome.Succeeded(fa) =>
-                  fa.flatMap { resp =>
-                    Resource
-                      .eval {
-                        span.addAttributes(
-                          response(resp, allowedResponseHeaders) ++
-                            additionalResponseAttributes(resp.responsePrelude)
-                        )
-                      }
-                      .mapK(trace)
-                  }
-                case Outcome.Errored(e) =>
-                  Resource.eval(span.recordException(e)).mapK(trace)
```

Comment:
`Tracer[F].(...).build.resource` already sets the status upon finalization of a resource.